external/chromium/src/base/allocator/partition_allocator/thread_isolation/thread_isolation.cc

```diff
@@ -52,30 +52,39 @@ int MprotectWithThreadIsolation(void* addr,
 #endif
 }
 
+void WriteProtectThreadIsolatedGlobals(
+    std::array<ThreadIsolationOption, kNumCompartments> &thread_isolations) {
+  for (auto thread_isolation : thread_isolations) {
+    WriteProtectThreadIsolatedGlobals(thread_isolation);
+  }
+}
+
 void WriteProtectThreadIsolatedGlobals(ThreadIsolationOption thread_isolation) {
-  WriteProtectThreadIsolatedVariable(thread_isolation,
-                                     PartitionAddressSpace::setup_);
+  // TODO(SJC): Protect compartment base addresses
+  // WriteProtectThreadIsolatedVariable(thread_isolation,
+  //                                    PartitionAddressSpace::setup_);
 
-  AddressPoolManager::Pool* pool =
-      AddressPoolManager::GetInstance().GetPool(kThreadIsolatedPoolHandle);
+  pool_handle handle = PoolHandleForCompartment(thread_isolation.compartment);
+  AddressPoolManager::Pool *pool =
+      AddressPoolManager::GetInstance().GetPool(handle);
   WriteProtectThreadIsolatedVariable(
-      thread_isolation, *pool,
-      offsetof(AddressPoolManager::Pool, alloc_bitset_));
```

Comment:
Did you intentionally drop the `offsetof` argument?

Reply:
Yes, we are protecting the entire pool, including the lock, on the same page. Previously the single thread isolation pool was carefully aligned to put the lock on the previous page, but that would double the number of pages we need and I don't really care about stats printing from out of the compartment (which is what the lock was used for outside the thread isolation thread).

Reply:
ok, that sounds reasonable. I'll have to take a closer look at the struct layout again.